test_scripts/API/Restructuring_OnResetTimeout/common_OnResetTimeout.lua

```diff
@@ -0,0 +1,463 @@
+      c.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", { })
+      c.getHMIConnection():SendNotification("TTS.Stopped")
+    end
+    RUN_AFTER(SpeakResponse, 2000)
+  end)
+  c.getMobileSession():ExpectResponse(cid, pRespParams)
+  :Timeout(pRespExpect)
+  :ValidIf(function()
+    local respTime = timestamp()
+    local timeBetweenRespAndNot = respTime - c.notificationTime
+    if timeBetweenRespAndNot >= pRespTime - 500 and timeBetweenRespAndNot <= pRespTime + 500 then
+      return true
+    else
+      return false, "Response is received in some unexpected time. Actual time is " .. timeBetweenRespAndNot ..
+      ". Ecpected time is " .. pRespTime
+    end
+  end)
+end
+
+function c.createInteractionChoiceSet()
+  local params = {
+    interactionChoiceSetID = 100,
+    choiceSet = {
+      {
+        choiceID = 111,
+        menuName = "Choice111",
+        vrCommands = { "Choice111" }
+      }
+    }
+  }
+  local corId = c.getMobileSession():SendRPC("CreateInteractionChoiceSet", params)
+  c.getHMIConnection():ExpectRequest("VR.AddCommand")
+  :Do(function(_, data)
+     c.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+  c.getMobileSession():ExpectResponse(corId, { success = true, resultCode = "SUCCESS" })
+end
+
+function c.PerformInteraction( pRespExpect, pRespTime, pResponseFunc, pParams, pRespParams )
+  if not pRespParams then pRespParams = { success = true, resultCode = "SUCCESS" } end
+  local params = {
+    initialText = "StartPerformInteraction",
+    interactionMode = "VR_ONLY",
+    interactionChoiceSetIDList = { 100 },
+    initialPrompt = {
+      { type = "TEXT", text = "pathToFile1" }
+    }
+  }
+  local corId = c.getMobileSession():SendRPC("PerformInteraction", params)
+  c.getHMIConnection():ExpectRequest("UI.PerformInteraction")
+  :Do(function(_, data)
+    c.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+  end)
+  c.getHMIConnection():ExpectRequest("VR.PerformInteraction", {
+    initialPrompt = params.initialPrompt
+  })
+  :Do(function(_, data)
+    pResponseFunc(data, pParams)
+  end)
+  c.getMobileSession():ExpectResponse(corId, pRespParams)
+  :Timeout(pRespExpect)
+  :ValidIf(function()
+    local respTime = timestamp()
+    local timeBetweenRespAndNot = respTime - c.notificationTime
+    if timeBetweenRespAndNot >= pRespTime - 500 and timeBetweenRespAndNot <= pRespTime + 500 then
+      return true
+    else
+      return false, "Response is received in some unexpected time. Actual time is " .. timeBetweenRespAndNot ..
+      ". Ecpected time is " .. pRespTime
+    end
+  end)
+end
+
+function c.Slider( pRespExpect, pRespTime, pResponseFunc, pParams, pRespParams )
+  if not pRespParams then pRespParams = { success = true, resultCode = "SUCCESS" } end
+  local cid = c.getMobileSession():SendRPC("Slider",
+    {
+      numTicks = 7,
+      position = 1,
+      sliderHeader ="sliderHeader",
+      sliderFooter = { "sliderFooter" }
```

Comment:
Add `timeout = 1000` because in 006,007 scripts will applied default timeout in SDL + default timeout for Slider rpc 10000 + 10000 = 20000, in case timeout = 1000 in request general timeout will be 11000. Please also update expectation timeout for response expectation:
```
local sliderTimeout = 1000
  local cid = c.getMobileSession():SendRPC("Slider",
    {
      numTicks = 7,
      position = 1,
      sliderHeader ="sliderHeader",
      sliderFooter = { "sliderFooter" },
      timeout = sliderTimeout
    })
  EXPECT_HMICALL("UI.Slider", { appID = c.getHMIAppId() })
  :Do(function(_, data)
    pResponseFunc(data, pParams)
  end)
  c.getMobileSession():ExpectResponse(cid, pRespParams)
   :Timeout(pRespExpect + sliderTimeout)
```